### DISCUSSION TITLE
Unable to run on simulator

### DISCUSSION THREAD

**Discussion by curia-damiano** [peterfriese/Swift-Async-Await-Experiments#4] [General]
Hi @peterfriese ,
First of all, thank you very much for your blog post!
I am in the process of porting some async code from C#, and your blog post has given me the right direction to start.

Now, the code compiles, and I can also deploy it to a simulator with iOS 14.5 (clearly not my physical iPhone).
I am using XCode 12.5 beta 3, with Swift Development Snapshot of 21.03.

When I try to debug the app, I get the following error: Could not launch "myAppName". LLDB provided no error string.
If I try with your demo app, clearly after changing the Team, it compiles and deploys as well, starts but then closes immediately (but XCode doesn't detect any crash).

Do you have similar experiences?
I suspect I could still miss something in my setup... do you have any idea?

Regards, Damiano

**Comment by peterfriese** [comment]
Hi Damiano,

thanks for reaching out! I had the same experience - the debugger doesn't connect. In my article I've described a workaround (i.e. turn off debugging). Check it out: https://peterfriese.dev/async-await-in-swiftui/ (there's also a video: https://youtu.be/K6UzzJOXYsA)

Cheers,
Peter

**Reply by curia-damiano** [reply]
Thank you Peter for your quick answer!
In this case my app behaves like yours: it starts but closes immediately.

Async/await is a very welcome news in Swift but currently:
- in case of some syntactical errors, the compiler crashes without any clear error;
- no possible to run apps on physical devices and submit apps to the Store;
- no possible to debug in the simulator;
- for some reason, no possible to start the app in the simulator.

For sure it will be a very widely used feature of the language (like it is now in C# and other languages), but currently it is still a bit immature.

But again, thank you very much for your article, video and help!

**Reply by peterfriese** [reply]
As this is still under development, some issues are to be expected. I guess this will be released in Swift 5.5. In the meantime, we can follow the development of this feature by downloading the latest builds of the compiler toolchain.
